Sentinel/Classes/Core/Internal/SentinelInternal.swift

```diff
@@ -35,82 +35,139 @@ extension Sentinel {
     }
 }
 
-// MARK: - Private extension
+enum Tabs {
+    case device
+    case application
+    case tools(items: [Tool])
+    case preferences(items: [OptionSwitchItem])
+    case performance
 
-private extension Sentinel {
-    func createTabBarControllers(
-        with tools: [Tool],
-        preferences: [OptionSwitchItem],
-        viewController: UIViewController
-    ) -> [UIViewController] {
-        return [
-            createDeviceViewController(),
-            createApplicationViewController(),
-            createToolsController(with: tools),
-            createPreferencesViewController(preferences),
-            createPerformanceViewController()
-        ]
+    var index: Int {
+        switch self {
+        case .device:
+            return 0
+        case .application:
+            return 1
+        case .tools:
+            return 2
+        case .preferences:
+            return 3
+        case .performance:
+            return 4
+        }
```

Comment:
I understand the logic behind it, but I would suggest not using indexes at all - everything will be calculated dynamically.

First of all, I would suggest changing the method 

```swift
func createTabBarControllers(
    with tools: [Tool],
    preferences: [OptionSwitchItem],
    viewController: UIViewController
) -> [UIViewController] { ... }
```

into

```swift
func createTabItems(
    with tools: [Tool],
    preferences: [OptionSwitchItem],
    viewController: UIViewController
) -> [TabItem] {
    return [
        TabItem(tab: .device),
        TabItem(tab: .application),
        TabItem(tab: .tools(items: tools)),
        TabItem(tab: .preferences(items: preferences)),
        TabItem(tab: .performance)
    ]
}

```

This way, we can get information about items (not just view controllers).

And now, in the `SentinelInternal.swift`, we can calculate the `preselectedIndex` (safely) and also configure view controller;

```swift
    func present(
        tools: [Tool],
        preferences: [OptionSwitchItem],
        on viewController: UIViewController
    ) {
        let tabItems = createTabItems(
            with: tools,
            preferences: preferences,
            viewController: viewController
        )
        let preselectedTabIndex = preselectedTabIndex(
            for: .tools(items: []),
            tabItems: tabItems
        )

        ...
        
        tabBarController.setupViewControllers(
            with: tabItems.map(\.viewController),
            preselectedIndex: preselectedTabIndex
        )
        ...
```